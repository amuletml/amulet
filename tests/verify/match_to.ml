external val (+) : int -> int -> int = "nil"

let _ = function
| { x, y } -> x + y

(* Patterns with non-total patterns should not be warned. *)
let _ = function
| true -> ()

(* Don't warn when we're already of this form *)
let _ = fun { x, y } -> x + y

let f g =
  match g () with
  | { x, y } -> x + y

let f g =
  match g () with
  | true -> ()

let f g =
  let { x, y } = g ()
  x + y
