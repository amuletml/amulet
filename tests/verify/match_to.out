match_to.ml[3:9 ..3:16]: note (N3007)
  match with a single arm can be rewritten using let.
  │ 
3 │ let _ = function
  │         ^^^^^^^^
  • Note: Replace with let { x = x, y = y } = …
match_to.ml[7:9 ..7:16]: warning (W3006)
  Non-exhaustive patterns in expression
  │ 
7 │ let _ = function
  │         ^^^^^^^^
  • Note: The following patterns are not covered
      false
match_to.ml[14:3 ..14:7]: note (N3007)
  match with a single arm can be rewritten using let.
   │ 
14 │   match g () with
   │   ^^^^^
  • Note: Replace with let { x = x, y = y } = …
match_to.ml[18:3 ..18:7]: warning (W3006)
  Non-exhaustive patterns in expression
   │ 
18 │   match g () with
   │   ^^^^^
  • Note: The following patterns are not covered
      false
